src/ws/ws-server.js

```diff
@@ -217,13 +219,25 @@ function processNewConnection(ws) {
         });
 }
 
+function createNewParticipant(message) {
+    return UserService.create(message.userData)
+              .then((user) => {
+                  log('[ws-server]', 'Create new user');
+
+                  const { uid } = user;
+                  phoenix.send(stateService.participantCreated(uid));
+              });
+}
+
 function processServerMessage(message) {
     // TODO: move participant validation here
     switch (message.name) {
         case MESSAGE_NAME.participantJoined:
             return participantIdentified(message.participantId, message.sessionId, message.role);
         case MESSAGE_NAME.solutionEvaluated:
             return solutionEvaluated(message);
+        case MESSAGE_NAME.createParticipant:
+            return createNewParticipant(message);
```

Comment:
IMHO `return createNewParticipant(message.userData);` is better - `createNewParticipant` doesn't need the whole message